src/orchestratord/src/controller/materialize/resources.rs

```diff
@@ -485,6 +532,76 @@ fn create_network_policies(
     network_policies
 }
 
+fn create_orchestratord_network_policies(
+    config: &super::Args,
+    mz: &Materialize,
+    orchestratord_namespace: &str,
+) -> Vec<NetworkPolicy> {
+    let mut network_policies = Vec::new();
+    if config.network_policies.internal_enabled {
+        let environmentd_label_selector = LabelSelector {
+            match_labels: Some(
+                mz.default_labels()
+                    .into_iter()
+                    .chain([(
+                        "materialize.cloud/app".to_owned(),
+                        mz.environmentd_service_name(),
+                    )])
+                    .collect(),
+            ),
+            ..Default::default()
+        };
+        let orchestratord_label_selector = LabelSelector {
+            match_labels: Some(
+                config
+                    .orchestratord_pod_selector_labels
+                    .iter()
+                    .cloned()
+                    .map(|kv| (kv.key, kv.value))
+                    .collect(),
+            ),
+            ..Default::default()
+        };
+        network_policies.extend([NetworkPolicy {
+            // can't use managed_resource_meta here because owner references
+            // don't work cross-namespace - we need to be sure to manually
+            // do the cleanup for this resource
+            metadata: ObjectMeta {
+                namespace: Some(orchestratord_namespace.to_string()),
+                name: Some(mz.name_prefixed("egress-to-environmentd")),
+                labels: Some(mz.default_labels()),
+                ..Default::default()
+            },
+            spec: Some(NetworkPolicySpec {
+                egress: Some(vec![NetworkPolicyEgressRule {
+                    to: Some(vec![NetworkPolicyPeer {
+                        namespace_selector: Some(LabelSelector {
+                            match_labels: Some(btreemap! {
+                                "kubernetes.io/metadata.name".into()
+                                    => mz.namespace(),
+                            }),
+                            ..Default::default()
+                        }),
```

Comment:
If we leave off the namespace selector, and just use a simplified pod selector of ```
{
    "materialize.cloud/app".to_owned() => mz.environmentd_service_name(),
}
```
could we instead have this just be installed once by the helm chart, allowing egress to environmentd in all namespaces?

It feels a bit gross creating resources across multiple namespaces, when the customer only created the Materialize CR in one. That is likely to break expectations, and it makes our code messier.